pkg/nfd-master/nfd-master.go

```diff
@@ -738,21 +793,22 @@ func authorizeClient(c context.Context, checkNodeName bool, nodeName string) err
 	return nil
 }
 
-func (m *nfdMaster) processNodeFeatureRule(features *nfdv1alpha1.Features) (map[string]string, []corev1.Taint) {
+func (m *nfdMaster) processNodeFeatureRule(features *nfdv1alpha1.Features) (map[string]string, map[string]string, []corev1.Taint) {
```

Comment:
Would it be possible to use the ExtendedResources type here? Like

```suggestion
func (m *nfdMaster) processNodeFeatureRule(features *nfdv1alpha1.Features) (map[string]string, ExtendedResources, []corev1.Taint) {
```

...if so then we'd probably want to use the Labels type for labels, too...